File: internal/ipc/transport/limits.go
//go:build linux

package transport

import "time"

const (
	// MaxConnections is the maximum number of concurrent connections allowed.
	MaxConnections = 100
	// ReadTimeout is the timeout for reading from a connection.
	ReadTimeout = 5 * time.Second
	// WriteTimeout is the timeout for writing to a connection.
	WriteTimeout = 5 * time.Second
	// MaxMsgSize is the maximum size of a message.
	MaxMsgSize = 1024 * 1024 // 1MB
)

File: internal/ipc/transport/limits.rs
//! Transport-wide limits.

/// Maximum number of concurrent connections the server accepts.
#[allow(non_upper_case_globals)]
pub const MaxConnections: usize = 100;
/// Per-request read timeout.
#[allow(non_upper_case_globals)]
pub const ReadTimeout: std::time::Duration = std::time::Duration::from_secs(5);
/// Per-request write timeout.
#[allow(non_upper_case_globals)]
pub const WriteTimeout: std::time::Duration = std::time::Duration::from_secs(5);
/// Hard ceiling on a single message. A peer sending more is rejected.
#[allow(non_upper_case_globals)]
pub const MaxMsgSize: usize = 1024 * 1024;
